### ISSUE TITLE
(UAT) Hidden Button

### ISSUE THREAD

**Issue by KKaselkar** [unicef/etools-issues#760]
Blocked & Marked for deletion Partner imported from vision displays in normal list without enabling the show hidden button. They should only appear when the show hidden button is active.

**Comment by ewheeler** [comment]
is this a duplicate of #453?

please check for existing issues instead of creating multiple issues describing the same thing

**Comment by KKaselkar** [comment]
![untitled](https://user-images.githubusercontent.com/32198614/30930707-7cdca0b6-a390-11e7-9e15-875235569750.png)

**Comment by KKaselkar** [comment]
If it is a duplicate then I will delete this issue. Give me sometime to add the rest and I will review the duplicates if exists.

**Comment by ewheeler** [comment]
@KKaselkar please check for existing issues BEFORE creating new ones and add any additional info there

**Comment by KKaselkar** [comment]
Issue 453 doesn't say anything about block partners and this issue is related to TRIP and not PMP Redesign.

**Comment by ewheeler** [comment]
sounds a lot like #689

again, please search for existing issues!

**Comment by KKaselkar** [comment]
No, its totally different. 689 doesn't even say anything about hide button.

**Comment by ewheeler** [comment]
ok. sounds very much related to me. again, without enough information to reproduce.. its hard to be sure

**Comment by KKaselkar** [comment]
Steps to reproduce this error:
1) Add new Partner ( e.g. number :2500217801 which is blocked in vision)
2) displays in active list without enabling show hidden button

**Comment by mpawlowski85** [comment]
Indeed - there seems to be the odd partner that is blocked and/or marked for deletion, that displays on the list, even though the show hidden toggle is n the off position. Example from Pakistan:

**Comment by daniel-tabarcea** [comment]
I looked into it and i've found that `ASIAN DISASTER PREPAREDNESS CENTER` is in the list twice, one is hidden and one isn't. 

![same partner twice](https://user-images.githubusercontent.com/17467562/31129693-025f282c-a85e-11e7-9a31-ba868c8ec432.png)

Is this normal?

**Comment by ShrinivasPuranik** [comment]
This is only for manually added partners.

**Comment by mpawlowski85** [comment]
This requires the new API for partner data to be used.

**Comment by domdinicola** [comment]
We are trying to integrate the new API:
the API is providing 3 fields

PURCHASING_BLOCK	VARCHAR2	Purchasing Block	 
POSTING_BLOCK	VARCHAR2	Posting Block	 
BLOCK_FUNCTION	VARCHAR2	Block Function

we need to know which field to use and the mapping since all three are VARCHAR2

**Comment by mpawlowski85** [comment]
@domdinicola we will have to use 
POSTING_BLOCK	VARCHAR2	Posting Block
thnaks

**Comment by domdinicola** [comment]
trying to change the endpoint there are some problems:

for the same country (Jordan) we have different number of results:
GetPartnerDetailsInfo gives 128 results
GetPartnershipInfo gives 1299 results

moreover

CSO_TYPE is not always there
RISK_RATING is not always there
CORE_VALUE_ASSESSMENT_DT is 19-JUL-13 not datetime and not always there
DATE_OF_ASSESSMENT is 19-JUL-13 not datetime and not always there
PHONE_NUMBER is not always there
TYPE_OF_ASSESSMENT is not always there
STREET is not always there
EMAIL is not always there
CITY is not always there

we also need a list of required keys in order to "accept" the record as valid
